Release/include/cpprest/string_utils.h

```diff
@@ -0,0 +1,441 @@
+#include <xlocale.h>
+#endif
+#endif
+
+/// Various utilities for string conversions and date and time manipulation.
+namespace utility
+{
+/// Functions for Unicode string conversions.
+namespace conversions
+{
+/// <summary>
+/// Converts a UTF-16 string to a UTF-8 string.
+/// </summary>
+/// <param name="w">A two byte character UTF-16 string.</param>
+/// <returns>A single byte character UTF-8 string.</returns>
+_ASYNCRTIMP std::string __cdecl utf16_to_utf8(utf16string_view w);
+
+/// <summary>
+/// Converts a UTF-8 string to a UTF-16
+/// </summary>
+/// <param name="s">A single byte character UTF-8 string.</param>
+/// <returns>A two byte character UTF-16 string.</returns>
+_ASYNCRTIMP utf16string __cdecl utf8_to_utf16(utility::nstring_view_t s);
+
+/// <summary>
+/// Converts a ASCII (us-ascii) string to a UTF-16 string.
+/// </summary>
+/// <param name="s">A single byte character us-ascii string.</param>
+/// <returns>A two byte character UTF-16 string.</returns>
+_ASYNCRTIMP utf16string __cdecl usascii_to_utf16(utility::nstring_view_t s);
+
+/// <summary>
+/// Converts a Latin1 (iso-8859-1) string to a UTF-16 string.
+/// </summary>
+/// <param name="s">A single byte character UTF-8 string.</param>
+/// <returns>A two byte character UTF-16 string.</returns>
+_ASYNCRTIMP utf16string __cdecl latin1_to_utf16(utility::nstring_view_t s);
+
+/// <summary>
+/// Converts a Latin1 (iso-8859-1) string to a UTF-8 string.
+/// </summary>
+/// <param name="s">A single byte character UTF-8 string.</param>
+/// <returns>A single byte character UTF-8 string.</returns>
+_ASYNCRTIMP utf8string __cdecl latin1_to_utf8(utility::nstring_view_t s);
+
+/// <summary>
+/// Converts to a platform dependent Unicode string type.
+/// </summary>
+/// <param name="s">A single byte character UTF-8 string.</param>
+/// <returns>A platform dependent string type.</returns>
+#ifdef _UTF16_STRINGS
+_ASYNCRTIMP utility::string_t __cdecl to_string_t(std::string&& s);
+#else
+inline utility::string_t&& to_string_t(std::string&& s) { return std::move(s); }
+#endif
+
+/// <summary>
+/// Converts to a platform dependent Unicode string type.
+/// </summary>
+/// <param name="s">A two byte character UTF-16 string.</param>
+/// <returns>A platform dependent string type.</returns>
+#ifdef _UTF16_STRINGS
+inline utility::string_t&& to_string_t(utf16string&& s) { return std::move(s); }
+#else
+_ASYNCRTIMP utility::string_t __cdecl to_string_t(utf16string&& s);
+#endif
+/// <summary>
+/// Converts to a platform dependent Unicode string type.
+/// </summary>
+/// <param name="s">A single byte character UTF-8 string.</param>
+/// <returns>A platform dependent string type.</returns>
+#ifdef _UTF16_STRINGS
+_ASYNCRTIMP utility::string_t __cdecl to_string_t(utility::nstring_view_t s);
+#else
+inline const utility::string_t& to_string_t(const std::string& s) { return s; }
+#if CPPREST_USE_STRING_VIEWS
+inline utility::string_view_t to_string_t(utility::nstring_view_t s) { return s; }
```

Comment:
All overloads of `to_string_t` should return a string object, not a string view, since the expectations of the function is to return a string object:
```
/// <param name="s">A single byte character UTF-8 string.</param>
```

If you sometimes return a string object and sometimes a string view it may lead to unexpected disasters, e.g.:
```
auto text = to_string_t(getSomeText());
```
If `getSomeText()` returns a string object or a string view, you're OK.
But if it returns something convertible to string view, after that statement the returned temporary is destroyed, `text` is dangling now and you're most certainly screwed.